tests/test_notebooks.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+import sys
+from pathlib import Path
+
+import papermill as pm
+import pytest
+
+
+@pytest.fixture()
+def common_kwargs(tmpdir):
+    outputnb = tmpdir.join("output.ipynb")
+    return {
+        "output_path": str(outputnb),
+        "kernel_name": f"python{sys.version_info.major}",
+        "progress_bar": False,
+    }
+
+
+def test_intro(common_kwargs):
+    execution_dir = Path.cwd() / "docs" / "usage"
+    pm.execute_notebook(execution_dir / "intro.ipynb", **common_kwargs)
+
+
+def test_vector_design_prototype(common_kwargs):
+    execution_dir = Path.cwd() / "docs" / "usage"
+    pm.execute_notebook(
+        execution_dir / "vector_design_prototype.ipynb", **common_kwargs
+    )
```

Comment:
This notebook currently fails as there is a problem with the API used in the notebook (so what Issue #247 wanted to check for).

```pytb
$ pytest tests/test_notebooks.py -k test_vector_design_prototype
E           papermill.exceptions.PapermillExecutionError: 
E           ---------------------------------------------------------------------------
E           Exception encountered at "In [3]":
E           ---------------------------------------------------------------------------
E           TypeError                                 Traceback (most recent call last)
E           Cell In [3], line 1
E           ----> 1 x, y, z, t = tree.arrays(["Muon_Px", "Muon_Py", "Muon_Pz", "Muon_E"], outputtype=tuple)
E                 2 offsets = ak.layout.Index64(x.offsets)
E                 3 content = ak.layout.RecordArray(
E                 4     {
E                 5         "x": ak.layout.NumpyArray(x.content.astype(np.float64)),
E              (...)
E                10     parameters={"__record__": "LorentzXYZT"},
E                11 )
E           
E           TypeError: HasBranches.arrays() got an unexpected keyword argument 'outputtype'
```

Reply:
Ah this is because the notebook was written using `uproot3`: https://github.com/scikit-hep/uproot5/blob/fb978c505eca58347a740122c5cfdf27fd09353a/docs-sphinx/uproot3-to-4.rst#reading-arrays

Reply:
@Saransh-cpp @henryiii @jpivarski I'm a bit confused how this notebook was ever written (I'm trying to get the old version to run before attempting to update to newer APIs &mdash; maybe a mistake) but as `awkward1` and `uproot3` were used here then I need

```
python -m pip install --upgrade awkward 1'uproot<4.0.0'
```

which allows for

```python
tree = uproot.open(data_path("uproot-HZZ.root"))["events"]
x, y, z, t = tree.arrays(["Muon_Px", "Muon_Py", "Muon_Pz", "Muon_E"], outputtype=tuple)
```

but then

```python
offsets = ak.layout.Index64(x.offsets)
```

raises

```pytb
AttributeError: module 'awkward1' has no attribute 'layout'
```

Have any thoughts here how this ever got evaluated when @henryiii added in in PR #59? Or should this just be given up on and should the whole thing be rewritten?

Reply:
Thanks Jim!

Reply:
I started working on this notebook but couldn't complete the migration - https://github.com/Saransh-cpp/vector/blob/update-vector-design-prototype/docs/usage/vector_design_prototype.ipynb.

(I was stuck at the `numba` part, and there was some discussion about it here - https://iris-hep.slack.com/archives/C01BBQDEB6X/p1664214223987299, but I couldn't pick up the leftover work because of college.)

We would also need to remove this notebook's reference from awkward's documentation - https://awkward-array.readthedocs.io/en/latest/ak.behavior.html#complete-example. (I can create a quick PR once this is merged 😄)

Reply:
Yeah that would be good to remove it and then readd a modern version of it later as needed.

Reply:
Eventually maybe we can re-add it, it's a great resource for how to write a numba extension (well, it was). Fine to take it out for now.